### PR TITLE
MongoDB Atlas vector store

### PR DESCRIPTION
**Pull request by hongyishi**
# Description

We want to collaborate with MongoDB Atlas Search to show how we can implement the Atlas

## Type of Change
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Added to MongoDB nb an atlas search example.

# Suggested Checklist:

- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/examples/data_connectors/MongoDemo.ipynb [hunk 1]
```diff
@@ -84,6 +85,55 @@
...
+            "source": [
+                "# Provide URI to constructor, or use environment variable\n",
+                "import pymongo\n",
+                "from llama_index.vector_stores.mongodb import MongoDBVectorStore\n",
+                "from llama_index.indices.vector_store.base import VectorStoreIndex\n",
+                "from llama_index.storage.storage_context import StorageContext\n",
+                "from llama_index.readers.file.base import SimpleDirectoryReader\n",
+                "\n",
+                "\n",
+                "# mongo_uri = os.environ[\"MONGO_URI\"]\n",
```
Comment (logan-markewich) [inline docs/examples/data_connectors/MongoDemo.ipynb:127]: nit: this should probably be broken into a few different cells

File: llama_index/vector_stores/mongodb.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        # delete by filtering on the doc_id metadata
+        self._collection.delete_one(filter={"id": ref_doc_id}, **delete_kwargs)
```
Comment (logan-markewich) [inline llama_index/vector_stores/mongodb.py:131]: does this actually delete by ref_do_id? It seems like the `id` field has the node_id

**Review by logan-markewich — APPROVED** [review]
Nice, looking good! I love how easy it is to add vector stores (just a single file)

Assuming the delete works as intended and is tested, it looks good to go

File: llama_index/vector_stores/mongodb.py (hunk 2 above)
Comment (hongyishi) [reply llama_index/vector_stores/mongodb.py]: Should be fixed now

File: docs/examples/data_connectors/MongoDemo.ipynb (hunk 1 above)
Comment (hongyishi) [reply docs/examples/data_connectors/MongoDemo.ipynb]: Fixed

File: llama_index/vector_stores/mongodb.py (hunk 2 above)
Comment (logan-markewich) [reply llama_index/vector_stores/mongodb.py]: Yea this makes more sense, thanks!

File: docs/examples/data_connectors/MongoDemo.ipynb (hunk 1 above)
Comment (logan-markewich) [reply docs/examples/data_connectors/MongoDemo.ipynb]: Looks good!

**Comment by logan-markewich** [comment]
Oh one last thing, let's add this to the changelog

**Review by jerryjliu — COMMENTED** [review]
sweet! 


nit: the notebook should probably be in docs/examples/vector_stores, not docs/examples/data_connectors

**Pull request merged**